src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -798,12 +709,20 @@ public int getCurrentPoolSize() {
      * @return current number of workers executing tasks
      */
     public int getCurrentRunningCount() {
-      synchronized (workersLock) {
-        int workerOutQty = currentPoolSize - availableWorkers.size();
-        if (waitingForWorker || currentPoolSize == 0) {
-          return workerOutQty;
+      int startPoolSize = currentPoolSize.get();
+      while (true) {
+        int idleWorkerCount = 0;
+        Worker idleWorker = this.idleWorker.get();
+        while (idleWorker != null) {
+          idleWorkerCount++;
+          idleWorker = idleWorker.nextIdleWorker;
+        }
+      
+        if (currentPoolSize.get() == startPoolSize) {
+          // TODO - do we want to re-check if the idle queue changed during computation
```

Comment:
I want to at least have a consistent pool size, otherwise we could return a negative number here (if they shrink the pool while this is being calculated, and idle workers were killed).

But I agree, I am leaning towards just removing this TODO note and calling it good.